eng/templates/stages/release.yml

```diff
@@ -304,10 +302,22 @@ stages:
             echo "Announcement title: $title"
             echo "Announcement body: $body"
           else
-            echo "Submitting announcement."
-            announcement_url=$( gh api graphql -F RepoId=$(RepoId) -F categoryId=$(DiscussionCategoryId) -F body="$body" -F title="$title" -f query="$query" --template '{{.data.createDiscussion.discussion.url}}' )
+            # Checking for an already existing announcement
+            query='query { repository(name: "${{ variables.announcementRepo }}", owner: "${{ variables.announcementOrg }}") { discussions(last: 100, categoryId: "$(DiscussionCategoryId)", orderBy: {field: UPDATED_AT, direction: DESC}) { edges { node { title url } } } } }'
+
+            response=$(gh api graphql -f query="$query")
+            discussion=$(echo "$response" | jq -r '.data.repository.discussions.edges[] | select(.node.title == "'"$title"'") | .node')
```

Comment:
Consider more descriptive variable names:

* `query` => `recentAnnouncementsQuery`
* `response` => `recentAnnouncementsResponse`
* `discussion` => `discussionWithSameTitle`

Reply:
I will put both steps - getting discussion category and submitting the announcement in a script that's just easy to test etc. This shouldn't take long